### PR TITLE
Correct how the tags helper gets the text for tags

### PR DESCRIPTION
**Pull request by iainbeeston**
# Description

Fixes the tags test helper so it returns the text of each visible tag.

The tags helper was getting the `label` attribute of `.tagify__tag` elements before but so far as I can see `.tagify__tag` elements do not have a label. Because of that the tags helper was always returning an array of `nil` values (one for each tag). Instead I've changed them to get the text of the tag.

I'm not sure why this wasn't causing the avo test suite to fail before, as there is a test for this helper and the other tags helpers (which return the result of the tags helper) but this seems to work for both the avo test suite and my own code (and matches what I can see when I try avo myself in the browser). It also matches what the `wait_for_tag_to_appear` helper does (ie. looks for tag text, not label).

# Fixes # (issue)

The tags test helper was always returning an array of `nil` values (one for each tag).

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the [documentation](https://github.com/avo-hq/avodocs)
- [ ] I have added tests that prove my fix is effective or that my feature works

## Manual review steps

So long as the avo test suite still passes this should be ok.

Manual reviewer: please leave a comment with output from the test if that's the case.

### CODE REVIEW

File: spec/system/avo/test_helpers_spec.rb [hunk 1]
```diff
@@ -222,6 +222,14 @@
         expect(remove_tag(field: :tags, tag: "one")).to eq ["three"]
         expect(tag_suggestions(field: :tags, input: "")).to eq ["one", "two"]
       end
+
+      let!(:post) { create :post, tag_list: ["test", "five", "seven"] }
+
+      it "verify tags" do
+        visit avo.edit_resources_post_path(post)
+
+        expect(tags(field: :tags)).to eq ["test", "five", "seven"]
```
Comment (iainbeeston) [inline spec/system/avo/test_helpers_spec.rb:231]: It looks like in the test this is coming out as `[“seven”, “five”, “test”]`

Comment (Paul-Bob) [reply spec/system/avo/test_helpers_spec.rb]: True, it was passing locally, but the tags field on the post has enforced suggestions so it's flaky... Just changed it to `["one", "two"]` should work as expected not

**Review by Paul-Bob — APPROVED** [review]
> I'm not sure why this wasn't causing the avo test suite to fail before, as there is a test for this helper and the other tags helpers (which return the result of the tags helper) 

The test case wasn't covering it enough, thanks for this contribution @iainbeeston .

I've committed a test for it.

**Comment by Paul-Bob** [comment]
@iainbeeston this helper seems a bit unpredictable, it returns tags in different orders each time. Any ideas why?

**Comment by iainbeeston** [comment]
@Paul-Bob I think that's happening because acts as taggable does not preserve the order of tags by default. For this test to work the way you want you'll need to update `Post` from using `acts_as_taggable_on :tags` to `acts_as_ordered_taggable_on :tags` (documentation is [here](https://github.com/mbleigh/acts-as-taggable-on?tab=readme-ov-file#usage))

**Comment by iainbeeston** [comment]
The other option would be to use the `contains_exactly` matcher on the list of tags returned.

**Comment by Paul-Bob** [comment]
That was it, thanks for looking into this @iainbeeston !

**Pull request merged**